userspace/falco/application.cpp

```diff
@@ -172,13 +172,14 @@ bool application::run(std::string &errstr, bool &restart)
 	// dependencies are honored (e.g. don't process events before
 	// loading plugins, opening inspector, etc.).
 	std::list<std::function<run_result()>> run_steps = {
+		std::bind(&application::load_config, this),
 		std::bind(&application::print_help, this),
 		std::bind(&application::print_version, this),
 		std::bind(&application::print_page_size, this),
 		std::bind(&application::print_generated_gvisor_config, this),
 		std::bind(&application::print_ignored_events, this),
 		std::bind(&application::print_syscall_events, this),
-		std::bind(&application::load_config, this),
+		std::bind(&application::require_config_file, this),
```

Comment:
Do i read this right that we now allow to run Falco without a config file to just print various infos?

Reply:
(it was already like this, right. Ignore me!)

Reply:
Correct! Basically:

- Falco loads its default config and applies all the CLI option overrides (`-o` option) right at start time. If a config file is specified, then Falco loads it instead of using the default config
- Alle the print actions are executed by not strictly requiring a config file (this was the original behavior), but will not have access to the configuration loaded up to that point (either the default one or the file-loaded one)
- Finally, at a certain point the Falco actions will require a non-default configuration to be loaded, thus will first check that the currently-loaded one came from a File

This was my best proposal for both allowing early actions to access the Falco config, while also preserving all the original behavior that required a config file to proceed.